### DISCUSSION TITLE
Redirecting blog.example.com to example.com/blog

### DISCUSSION THREAD

**Discussion by cereallarceny** [vercel/community#892] [Help] [answered]
I've been working on moving our blog from Squarespace to our Next.js app. The location of the blog originally was https://blog.raisefinancial.com, and it's now located at https://www.raisefinancial.com/blog. The deployment of the blog portion of our Next.js app went perfectly and the site is now live.

However, I'm now trying to see if I can get redirections/rewrites working correctly and have so far run into some issues.

**Goals**
- Be able to redirect/rewrite the root of the blog from https://blog.raisefinancial.com to https://www.raisefinancial.com/blog
- Be able to redirect/write all posts from https://blog.raisefinancial.com/posts/this-is-my-post to https://www.raisefinancial.com/blog/this-is-my-post

**Steps taken**
- A few months back when we spun up the Squarespace blog, I followed the setup instructions for directing `blog.raisefinancial.com` to Squarespace and this been working perfectly for many months. All of my DNS for this domain is handled by Vercel.
- Today, once I saw the deployment of the new blog was successful, removed the CNAME records for `blog.raisefinancial.com` which were originally pointing to Squarespace.
- I added a `vercel.json` file in the `apps/raise` directory of my monorepo (yes, I have configured Vercel to treat `apps/raise` as  the project root. This file contains the following:

```json
{
  "rewrites": [
    {
      "source": "/posts/:post*",
      "has": [
        {
          "type": "host",
          "value": "blog.raisefinancial.com"
        }
      ],
      "destination": "/blog/:post*"
    },
    {
      "source": "/",
      "has": [
        {
          "type": "host",
          "value": "blog.raisefinancial.com"
        }
      ],
      "destination": "/blog"
    }
  ]
}
```

- I had presumed that this would be enough to get the rewrite to trigger, however, now I'm seeing a `404: NOT_FOUND` error on the following URLS:
  - https://blog.raisefinancial.com/
  - https://blog.raisefinancial.com/posts/why-changing-the-way-you-invest-starts-now

**Expectations/Test Case**
I expect the following solution:

- https://blog.raisefinancial.com/ --> https://www.raisefinancial.com/blog
- https://blog.raisefinancial.com/posts/why-changing-the-way-you-invest-starts-now --> https://www.raisefinancial.com/blog/why-changing-the-way-you-invest-starts-now

**Resources**
- https://vercel.com/guides/can-i-redirect-from-a-subdomain-to-a-subpath

Any pointers here would be greatly appreciated!

**Comment by cereallarceny** [comment, marked as answer]
Got this to work by deleting the `vercel.json` file and moving it into my `next.config.js` file. I also needed to specify this as a redirect, not a rewrite. And I also needed to specify the destinations WITH the full URL:

```js
module.exports = (_phase, { defaultConfig }) => {
  const redirects = async () => {
    return [
      {
        source: '/posts/:post*',
        has: [
          {
            type: 'host',
            value: 'blog.raisefinancial.com',
          },
        ],
        destination: 'https://www.raisefinancial.com/blog/:post*',
        permanent: true,
      },
      {
        source: '/',
        has: [
          {
            type: 'host',
            value: 'blog.raisefinancial.com',
          },
        ],
        destination: 'https://www.raisefinancial.com/blog',
        permanent: true,
      },
    ];
  };

  const final = plugins.reduce((acc, plugin) => plugin(acc), {
    ...defaultConfig,
    ...config,
    redirects,
  });

  return final;
};
```

**Reply by amyegan** [reply]
I'm glad you got it working! Thanks for coming back to share your solution.

**Comment by Wormedi** [comment]
How to redirect my domain to vercel.app anyone help me

**Reply by amyegan** [reply]
Hi @Wormedi! You can [add a custom domain](https://vercel.com/docs/concepts/projects/domains/add-a-domain) to your Vercel project by modifying DNS records or changing nameservers.

In addition to the custom domain documentation, there is a guide available to help you through setup and troubleshooting: [How do I add a custom domain to my Vercel project?](https://vercel.com/guides/how-do-i-add-a-custom-domain-to-my-vercel-project)
